From a996bf7ead85ccd5824137f169a35c313b28fc21 Mon Sep 17 00:00:00 2001
From: database64128 <free122448@hotmail.com>
Date: Mon, 25 Aug 2025 16:02:24 +0800
Subject: [PATCH 101/133] std.net.Ip6Address: format numerical scope id
X-Developer-Signature: v=1; a=openpgp-sha256; l=1038; i=xtex@astrafall.org;
 h=from:subject; bh=it+bh++pr0xX3E91xBez2s3TgO1TqNoqbFHkAYmCgnc=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78rhmfk5//ULCRPWHjeH6+gNWX3Z03W2/HbKv0/r2v
 I9Tuo0KOyayMIhxMViKKbIUGTZ4s+qk84suK5eFmcPKBDJEWqSBgYGBgYWBLzcxr9RIx0jPVNtQ
 z9BQx0jHiIGLUwCmetMChv8JL06e0Z377Ieonp3czdTyil2qB34lcVmEbrbKeHODseQbw1+xN0d
 EnSfs9Fqrp3z4h0QTu/mynQsMKmpjD3sZ/zR5MYEVAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 377a8b2a3b408dad4392a97254a7a6bb2d06b200)
---
 lib/std/net.zig      | 3 +++
 lib/std/net/test.zig | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/std/net.zig b/lib/std/net.zig
index 03383b0f890d..9d70515c2ee1 100644
--- a/lib/std/net.zig
+++ b/lib/std/net.zig
@@ -747,6 +747,9 @@ pub const Ip6Address = extern struct {
                 try w.writeAll(":");
             }
         }
+        if (self.sa.scope_id != 0) {
+            try w.print("%{}", .{self.sa.scope_id});
+        }
         try w.print("]:{}", .{port});
     }
 
diff --git a/lib/std/net/test.zig b/lib/std/net/test.zig
index a7e6b465cbd0..bfafbe6044fa 100644
--- a/lib/std/net/test.zig
+++ b/lib/std/net/test.zig
@@ -73,7 +73,7 @@ test "parse and render IPv6 addresses" {
         "2001:db8::",
         "::1234:5678",
         "2001:db8::1234:5678",
-        "ff01::fb",
+        "ff01::fb%1234",
         "::ffff:123.5.123.5",
     };
     for (ips, 0..) |ip, i| {
-- 
2.52.0

